### PR TITLE
add stop to strategy-controllers

### PR DESCRIPTION
**Pull request by spreeker**
Hi in order to develop and test my strategies I wanted to keep track and plot / log all kinds of triggers when the bot is done.

When the bot is done running I'd like to have a callback into the strategy to cleanup / plot / log whatever. A stop function was missing. 

My personal reason: 
A strategy might want to buy and sell more then the current wallet / assetposition allows and the amount of successfull orders might be quite different from the actual orders wanted by the strategy. After a run I create a plot / statistics and compare that with the webinterface provided.

### CODE REVIEW

File: examples/strategies/emacross.go [hunk 1]
```diff
@@ -16,6 +16,9 @@ func (e CrossEMA) Timeframe() string {
 	return "1d"
 }
 
+// Stop can be used to close stuff of plot/log some strategy data
+func (e CrossEMA) Stop(df *model.Dataframe, broker exchange.Broker) {}
```
Comment (rodrigo-brito) [inline examples/strategies/emacross.go:20]: The name `Stop` can be misunderstood (eg. stop order, custom stop, stop signal). 
May we can rename Init the function `Init` to `Finish` and keep `Start` and `Finish`
Do you have any other name suggestions?

**Review by rodrigo-brito — CHANGES_REQUESTED** [review]

**Comment by rodrigo-brito** [comment]
Hi @spreeker, indeed it can be useful. But I do not like the name.
Another way to implement it:
After `bot.Run()` you can access all these objects and complete the process inside `main` function.

File: examples/strategies/emacross.go (hunk 1 above)
Comment (spreeker) [reply examples/strategies/emacross.go:20]: Agreed will change to finish. I called it stop to be consistent with the other stop functions scattered around the code.

**Comment by spreeker** [comment]
I looked into writing a stop function after the bot was finished. BUT. The strategy controller which contains all relevant candle information for each trading pair is already destroyed.  So I can call stop on the strategy but the strategy has not stored candle information / dataframe. The relevant data is in the strategy controller(s) Which are not accessible when the bot is finished.

File: examples/strategies/emacross.go (hunk 1 above)
Comment (spreeker) [reply examples/strategies/emacross.go]: renamed to finish.

**Comment by rodrigo-brito** [comment]
Hi @spreeker, I will remove the `Init` function in the next release.
The `strategy` init can be performed in a custom function, for example: `NewMyStrategy(setting Setting) Strategy`.
And now, the controller information is available with `bot.Controller()` for custom cleanups. 

I'm removing it because the interface forces users to create these functions and I think it will not be used in most cases.

**Comment by rodrigo-brito** [comment]
This information will be available with `ninjabot.Controller()`

**Pull request closed without merging**